lib/avro/builder/dsl.rb

```diff
@@ -89,6 +89,25 @@ def type(*)
         @last_object = super
       end
 
+      # Several variants:
+      # define_type(:foo, long)
+      # define_type(:bar, { namespace: 'com.example' }, long)
+      # define_type(:baz, 'com.example', long)
```

Comment:
I wonder if all of these variants that rely on positional args with not entirely obvious ordering will create confusion.